### PR TITLE
Agent skills for block development

### PR DESCRIPTION
**Pull request by hame-c8y**

### CODE REVIEW

File: .github/skills/develop-analytics-builder-block/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+```
+
+#### Available L10N validation keys
+
+| Key | Arguments | When to use |
+|-----|-----------|-------------|
+| `fwk_param_finite_positive_value` | `[paramName, actualValue]` | Value must be > 0.0 and finite |
+| `fwk_param_valid_range` | `[paramName, min, max, actualValue]` | Value must be within a bounded range (e.g. 0.0 to 1.0) |
+| `fwk_param_missing_required_value` | `[paramName]` | Required parameter not provided |
+| `fwk_param_invalid_value` | `[fieldName1, fieldName2]` | Related fields in conflict (e.g. lower > upper) |
```
Comment (hame-c8y) [inline .github/skills/develop-analytics-builder-block/SKILL.md:320]: @skom-c8y @gys-c8y I think some of these are made up. Given limited L10N support for custom blocks should we instruct the agent to not use these but plain Exceptions instead?

File: generated-blocks/ConstantRateOutput.mon [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+ * Constant Rate Output.
+ *
+ * Takes an arbitrary Value object as input and outputs the unchanged value,
+ * but at a modified rate controlled by the period parameter.
+ *
+ * The period parameter in seconds limits the output to one output per period.
+ * The period starts with the first received input. If no input is received in
+ * a time period and "Limit only" is disabled, the previous value is resent,
+ * ensuring exactly one output per period.
+ *
```
[the remark this answers is not in the document]
Comment (hame-c8y) [reply generated-blocks/ConstantRateOutput.mon:88]: Not true

File: .github/skills/build-analytics-builder-blocks/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+### If no test directory exists for a block
+
+Note this to the user — it means the block has no automated test coverage. Proceed with the build but warn that the block is untested.
+
+### If tests fail
+
+1. Read the failure output to identify the failing test and block.
+2. Fix the block `.mon` file.
```
Comment (C8Y-Mario) [inline .github/skills/build-analytics-builder-blocks/SKILL.md]: I don't think that the build/deployment step should try to fix something in the code. That will possibly change behaviour before deployment.

File: .github/skills/write-analytics-builder-tests/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+| Test | Purpose |
+|---|---|
+| `_001` | **Touch test** — deploys successfully, basic happy-path input → output |
+| `_002` ... | Edge cases, parameter variations, boundary values, inverse inputs |
+
+Each test should be focused — test one behaviour per file.
+
+---
+
+## Step 3 — Create the Test File
```
Comment (C8Y-Mario) [inline .github/skills/write-analytics-builder-tests/SKILL.md:41]: Isn't that part of the write-analytics-builder-tests Skill?

**Review by C8Y-Mario — CHANGES_REQUESTED** [review]
The PR seems to include a lot of unnecessary stuff (tests and blocks?)
I would expect that the Skills have clear definition responsibilities.

**Comment by hame-c8y** [comment]
> The PR seems to include a lot of unnecessary stuff (tests and blocks?) I would expect that the Skills have clear definition responsibilities.

These are blocks and tests generated by the skills. I included them for reference and also because they are useful for actual usage. They are in a separate folder to distinguish them from the manually coded blocks.

Per the agent skill specification, each of the skills has a description that describes applicability. Is anything lacking from these descriptions? The skills were picked up correctly by the agent in all cases during my tests.

File: .github/skills/build-analytics-builder-blocks/SKILL.md (hunk 3 above)
Comment (hame-c8y) [reply .github/skills/build-analytics-builder-blocks/SKILL.md]: Agreed. Changed the instructions to never try to fix the problem but to provide more guidance why the test may have failed.

File: .github/skills/write-analytics-builder-tests/SKILL.md (hunk 4 above)
Comment (hame-c8y) [reply .github/skills/write-analytics-builder-tests/SKILL.md:41]: Moved the instructions for the test-fix cycle here.

**Comment by C8Y-Mario** [comment]
Hey Harald, can you please remove all the blocks and tests from this PR?
I'm fine if you want them, but then each one should be a separate PR and discussion. I also don't think that it is a good idea to put them into a "generated-blocks" folder. 
If they are of value like the other ones and have been proper discussed and reviewed they should be in the same folder like the others. Just sticking the "generated"  label on them won't relieve us from ownership

**Comment by hame-c8y** [comment]
> Hey Harald, can you please remove all the blocks and tests from this PR? I'm fine if you want them, but then each one should be a separate PR and discussion. I also don't think that it is a good idea to put them into a "generated-blocks" folder. If they are of value like the other ones and have been proper discussed and reviewed they should be in the same folder like the others. Just sticking the "generated" label on them won't relieve us from ownership

I will not do that as I do not think it is a valuable use of my time to create 5 separate PRs.

**Comment by C8Y-Mario** [comment]
Sorry, if its not worth your time to create proper PRs then don't expect others to spend time to review generated code.
